### DISCUSSION TITLE
Quasar Capacitor google login issue

### DISCUSSION THREAD

**Discussion by sheikhatif** [quasarframework/quasar#11514] [CLI - Capacitor / Cordova modes]
Hello 
I am unable to integrate google login in quasar capacitor app. please help

**Comment by MilosPaunovic** [comment]
What exactly is troubling you? Anyway, you'll get better assistance if you post a Codepen reproduction of your issue or a code sample of some kind.

**Reply by sheikhatif** [reply]
Hello
I am using hello.js for social login. below are my login code for google login and its working fine in web application 
```js
authProvider(provider) {
      console.log(provider)
      var scope = provider == 'facebook' ? 'friends' : 'email+profile';
      console.log(scope)
      this.$hello(provider).login({ scope: scope})
        .then((res) => {
          if (this.$hello.getAuthResponse(provider) == null) {
            return
          }
          this.$hello(provider).api('me')
            .then((res) => {
              console.log(res)
              this.socialLogin(provider, res)
            })
        })      
 },

socialLogin: function(provider,response) {
      this.$q.loading.show();
      var self = this 
      this.$axios.post('/api/authenticate/social', {
        provider: provider,
        data:response
      })
      .then(function(response) {
        self.role = response.data.role
        self.$store.dispatch('users/saveToken', {
            token: response.data.token,
            remember: self.remember
        })
        self.$store.dispatch('users/fetchUser')
        if (self.role == 'employer') {
          self.getClientData()
        } else {
          self.getFreelancerData()
        }
        setTimeout(() => {
          self.$q.loading.hide();
          self.$router.push({ name: 'dashboard' })
        }, 1000); 
      })
      .catch(function (error) {
          self.$q.loading.hide();
          console.log(error)
      })
 },
```
but when i build my app into Capacitor andriod and try to login using google login button i get below error

![Screenshot_20211130-172802](https://user-images.githubusercontent.com/23072837/144048256-ba8c08bc-dc3f-4e06-9ec7-e7edc4f265ac.png)

**Comment by hawkeye64** [comment]
To me, it looks like you gave it `http://localhost` as the redirect URI. It needs an external URI (not localhost). And, I believe you have to be HTTPS as well.

**Reply by sheikhatif** [reply]
here is my redirect url configuration at google console
![image](https://user-images.githubusercontent.com/23072837/144214035-9e4f04aa-549e-462f-ad8f-cb8b7361cf1f.png)

**Reply by hawkeye64** [reply]
For every computer on earth, localhost will refer to a different IP address

**Reply by sheikhatif** [reply]
can you please help me out how to fix the issue

**Reply by hawkeye64** [reply]
You can use something like `ngrok` to give yourself an HTTPS (free version is good for up to 6 hours, but should work well enough here). When you set up ngrok, copy the https address it gives you and add as URI above instead of localhost. You will need to change this twice a day.

**Reply by sheikhatif** [reply]
my redirect url is https://public.eventon.ro I just add localhost for testing. I remove the localhost in redirect URL but still google login give me localhost error with https redirect url

**Comment by TobyMosque** [comment]
I'm not HelloJS expect, but i think web based authetication would work only in a hosted side (SPA, PWA, SSR, etc).

for capacitor/cordova apps, u'll need to install a plugin, here some instructions
https://capacitorjs.com/docs/plugins/cordova

In the past, i used that plugins to login using the google and the facebook oAuth services:
https://github.com/EddyVerbruggen/cordova-plugin-googleplus
https://github.com/cordova-plugin-facebook-connect/cordova-plugin-facebook-connect
https://devdactic.com/capacitor-google-sign-in/

**Comment by vik17ag** [comment]
As per my knowledge, you will have to keep http://localhost in google console because capacitor serves from localhost. I am also trying a solution where capacitor can use a different domain than localhost (like the production domain) so that i can remove the localhost from google cloud console ( off course for security reasons )

**Comment by Hashen110** [comment]
[OAuth: how to test with local URLs?](https://stackoverflow.com/questions/10456174/oauth-how-to-test-with-local-urls)
